hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java

```diff
@@ -194,6 +194,18 @@ public Pipeline createPipeline(ReplicationConfig replicationConfig,
     try {
       Pipeline pipeline = pipelineFactory.create(replicationConfig,
           excludedNodes, favoredNodes);
+      // In case in case if provided pipeline provider returns null.
```

Comment:
As this change is required due what we suspect is a bug in the RackScatterPlacementPolicy, where it returns less nodes than expected without giving an exception, I think we should make a change to the RackScatter policy too, so that it throws an exception indicating it is about to return successfully with less than expected. You could add a reference to this Jira in a comment so we can remove it later if we get to the bottom of the suspected bug.

Reply:
oh, yeah I remember we talked about this. Let me add that.

Reply:
agree with stephen. maybe we would better to check and handle this in that certain placement policy. if the placement policy can not choose enough datanodes for the new pipeline , an exception should be thrown from it.

Reply:
Moved the check into pipelineFactory#create